### PR TITLE
ARROW-12832: [JS] Write benchmarks in TypeScript

### PR DESCRIPTION
**Pull request by domoritz**

### CODE REVIEW

**Comment by domoritz** [comment]
* [ ] actually load the sources by default

**Comment by kou** [comment]
Can we merge this?

**Comment by domoritz** [comment]
Since this pull request is part of https://github.com/apache/arrow/pull/10362 and @trxcllnt approved that one already, I think this can be merged as well.

File: js/tsconfig.json [hunk 1]
```diff
@@ -7,7 +7,8 @@
   "compilerOptions": {
     "target": "ESNEXT",
     "module": "commonjs",
-    "noEmit": true
+    "noEmit": true,
+    "esModuleInterop": true
```
Comment (TheNeuralBit) [inline js/tsconfig.json:11]: Just curious, is this necessary for this PR, or is it just something we want to do anyway (I see it's recommended [here](https://www.typescriptlang.org/tsconfig#esModuleInterop))

**Review by TheNeuralBit — APPROVED** [review]

File: js/tsconfig.json (hunk 1 above)
Comment (domoritz) [reply js/tsconfig.json:11]: Ahh, so this is the tsconfig file that editors like VSCode will pick up. We don't use it to compile things but just for type checking. esModuleInterop is needed for kleur in the perf tests. 

Screen Shot 2021-05-22 at 11 00 08

Comment (TheNeuralBit) [reply js/tsconfig.json:11]: Got it, thanks

Comment (domoritz) [reply js/tsconfig.json:11]: Having a single tsconfig for the editor that type checks everything is super convenient as we can see all references across the whole codebase including from tests and now benchmarks.

**Pull request closed without merging**